inject.js

```diff
@@ -132,56 +163,54 @@ function defineVideoController() {
     this.id = Math.random()
       .toString(36)
       .substr(2, 9);
-
-    // settings.speeds[] ensures that same source used across video tags (e.g. fullscreen on YT) retains speed setting
-    // this.speed is a controller level variable that retains speed setting across source switches (e.g. video quality, playlist change)
-    this.speed = 1.0;
-
+    storedSpeed = tc.settings.speeds[target.currentSrc];
     if (!tc.settings.rememberSpeed) {
-      if (!tc.settings.speeds[target.currentSrc]) {
-        tc.settings.speeds[target.currentSrc] = this.speed;
+      if (!storedSpeed) {
+        log(
+          "Overwriting stored speed to 1.0 due to rememberSpeed being disabled",
+          5
+        );
+        storedSpeed = 1.0;
       }
       setKeyBindings("reset", getKeyBindings("fast")); // resetSpeed = fastSpeed
     } else {
-      tc.settings.speeds[target.currentSrc] = tc.settings.lastSpeed;
+      log("Recalling stored speed due to rememberSpeed being enabled", 5);
+      storedSpeed = tc.settings.lastSpeed;
     }
 
-    target.playbackRate = tc.settings.speeds[target.currentSrc];
+    log("Explicitly setting playbackRate to: " + storedSpeed, 5);
+    target.playbackRate = storedSpeed;
 
     this.div = this.initializeControls();
 
     target.addEventListener(
       "play",
       (this.handlePlay = function(event) {
+        storedSpeed = tc.settings.speeds[event.target.currentSrc];
         if (!tc.settings.rememberSpeed) {
-          if (!tc.settings.speeds[target.currentSrc]) {
-            tc.settings.speeds[target.currentSrc] = this.speed;
+          if (!storedSpeed) {
+            log(
+              "Overwriting stored speed to 1.0 (rememberSpeed not enabled)",
+              4
+            );
+            storedSpeed = 1.0;
           }
+          // resetSpeed isn't really a reset, it's a toggle
+          log("Setting reset keybinding to fast", 5);
           setKeyBindings("reset", getKeyBindings("fast")); // resetSpeed = fastSpeed
         } else {
-          tc.settings.speeds[target.currentSrc] = tc.settings.lastSpeed;
-        }
-        target.playbackRate = tc.settings.speeds[target.currentSrc];
-      }.bind(this))
-    );
-
-    target.addEventListener(
-      "ratechange",
-      (this.handleRatechange = function(event) {
```

Comment:
If we're removing the listener it means our controller will no longer pickup ratechange events from other sources.. e.g. if someone uses built-in YT speed controls, the displayed speed will be out of sync. 

That seems like a regression. Thoughts>

Reply:
I actually didn't remove it, I just combined the two. The existing ratechange listener was placed in the DOM at the location of the video element (this seems wrong to say, but based on my observations this seems to be how an event propagates through the DOM - beginning at the root/document level then flowing down). This was too late, so I had to add another one ahead of it.

I now have combined them, so even if downstream ratechange listeners happen to be blocked by the edge case the controller is updated with the new speed. I will update the gherkin feature to explain the new behavior (the only change is very slight in the edge case)